app/presenters/hyrax/admin_set_presenter.rb

```diff
@@ -3,5 +3,22 @@ class AdminSetPresenter < CollectionPresenter
     def total_items
       ActiveFedora::SolrService.count("{!field f=isPartOf_ssim}#{id}")
     end
+
+    # AdminSet cannot be deleted if default set or non-empty
+    def disable_delete?
+      total_items > 0 || default_set?
+    end
+
+    # Message to display if deletion is disabled
+    def disabled_message
+      return I18n.t('hyrax.admin.admin_sets.delete.error_default_set') if default_set?
+      return I18n.t('hyrax.admin.admin_sets.delete.error_not_empty') if total_items > 0
+    end
+
+    private
+
+      def default_set?
```

Comment:
@mjgiarlo 👍  I completely missed your point about moving the method to the Class level (as part of the behavior) instead of at an instance level. Makes perfect sense now. I'll clean this up, as I agree this is a better route and would result in less duplication.